plugin/vscode/performance-writer.js

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2020  Matthew Glazar
+// See end of file for extended copyright information.
+
+const fs = require('fs');
+const path = require("path");
+const process = require("process");
+
+class PerformanceWriter {
+    
+    constructor (extensionContext) {
+        this._performanceTxtPath = path.join(process.cwd(), "test_performance.txt");
+        this._dataAttributes = {
+            columns:'timestamp,file-ref,file-name-or-duration,event-id'
+        };
+        this._fileName;
+        this._fileIndex;
+        this._extensionContext = extensionContext;
+    }
+
+    fileNameOptimizer (){
+        let currentSessionMap = this._extensionContext.globalState.get('qljsFileMap');
```

Comment:
Why isn't this an instance variable? What advantage is there to persisting this data in VS Code's globalState?

Reply:
Addressed in [this commit](https://github.com/lifeinData/quick-lint-js/4f4bdd46c6cdbe22e9b84ed2d69f9ed4d372fe5a)

Reply:
Put it as an instance variable in latest commit